### PR TITLE
Update Scala versions, ZIO, sbt and zio-sbt plugin

### PR DESCRIPTION
**Pull request by guizmaii**
Closes #878, closes #879

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,7 +24,6 @@ inThisBuild(
         run = Some("sbt ++${{ matrix.scala }} IntegrationTest/test")
       )
     ),
-    crossScalaVersions -= scala211.value,
```
Comment (guizmaii) [inline build.sbt:23]: zio-sbt doesn't provide support for Scala 2.11 anymore so we don't need this configuration anymore

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version = 1.8.3
+sbt.version = 1.9.4
```
Comment (mijicd) [inline project/build.properties:1]: Will it cause deprecation warnings (errors) due to the usage of integration test config?

Comment (guizmaii) [reply project/build.properties:1]: Yes, it does trigger warnings

File: build.sbt [hunk 3]
```diff
@@ -5,6 +5,10 @@ enablePlugins(ZioSbtEcosystemPlugin, ZioSbtCiPlugin)
 inThisBuild(
   List(
     name              := "ZIO Redis",
+    zioVersion        := "2.0.16",
+    scala212          := "2.12.18",
+    scala213          := "2.13.11",
+    scala3            := "3.3.0",
```
Comment (mijicd) [inline build.sbt:11]: These are yet to be upgraded in the plugin, right?

**Review by mijicd — APPROVED** [review]

File: build.sbt (hunk 3 above)
Comment (guizmaii) [reply build.sbt:11]: Not sure. I think I updated them recently in zio-sbt but I think you shouldn't have to wait for zio-sbt to update the Scala versions and make a new release for you to update yours

**Pull request merged**